dev/src/index.ts

```diff
@@ -1186,6 +1198,53 @@ export class Firestore implements firestore.Firestore {
     this.bulkWritersCount -= 1;
   }
 
+  /**
+   * Retrieves all descendant documents nested under the provided reference.
+   *
+   * @private
+   * @return {Stream<QueryDocumentSnapshot>} Stream of descendant documents.
+   */
+  // TODO(chenbrian): Make this a private method after adding recursive delete.
+  _getAllDescendants(
+    ref: CollectionReference | DocumentReference
+  ): NodeJS.ReadableStream {
+    // The parent is the closest ancestor document to the location we're
+    // deleting. If we are deleting a document, the parent is the path of that
+    // document. If we are deleting a collection, the parent is the path of the
+    // document containing that collection (or the database root, if it is a
+    // root collection).
+    let parentPath = ref._resourcePath;
+    if (ref instanceof CollectionReference) {
+      parentPath = parentPath.popLast();
+    }
+    const collectionId =
+      ref instanceof CollectionReference ? ref.id : ref.parent.id;
+
+    let query: Query<firestore.DocumentData> = new Query(
```

Comment:
```suggestion
    let query = new Query(
```

Reply:
Actually, i have to revert this change, since the tests are failing:
`Type 'Query<DocumentData>' is not assignable to type 'Query<unknown>'.`

Reply:
Hm, shouldn't it default to `DocumentData`? I will take a look.